DSharpPlus.SlashCommands/Converters/DefaultConverters.cs

```diff
@@ -0,0 +1,159 @@
+// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2022 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Threading.Tasks;
+using DSharpPlus.Entities;
+
+namespace DSharpPlus.SlashCommands.Converters
+{
+    public class StringConverter : ISlashArgumentConverter<string>
+    {
+        public ApplicationCommandOptionType OptionType { get; } = ApplicationCommandOptionType.String;
+
+        public Task<string> Convert(DiscordInteractionDataOption value, InteractionContext context)
+            => Task.FromResult(value.Value.ToString());
+    }
+
+    public class LongConverter : ISlashArgumentConverter<long>
+    {
+        public ApplicationCommandOptionType OptionType { get; } = ApplicationCommandOptionType.Integer;
+
+        public Task<long> Convert(DiscordInteractionDataOption value, InteractionContext context)
+            => Task.FromResult((long)value.Value);
+    }
+
+    public class NullableLongConverter : ISlashArgumentConverter<long?>
+    {
+        public ApplicationCommandOptionType OptionType { get; } = ApplicationCommandOptionType.Integer;
+
+        public Task<long?> Convert(DiscordInteractionDataOption value, InteractionContext context)
+            => Task.FromResult((long?)value.Value);
+    }
+
+    public class BoolConverter : ISlashArgumentConverter<bool>
+    {
+        public ApplicationCommandOptionType OptionType { get; } = ApplicationCommandOptionType.Boolean;
+        public Task<bool> Convert(DiscordInteractionDataOption value, InteractionContext context)
+            => Task.FromResult((bool)value.Value);
+    }
+
+    public class NullableBoolConverter : ISlashArgumentConverter<bool?>
```

Comment:
I think what should happen here (though this probably isbn't possible given the way you've written this) is that you should parse for nullable generic structs (`<T?> where T : struct`), and attempt to grab the parser for that type instead.

Reply:
This has been fixed.